### PR TITLE
SyncConfig has been extended by environment field

### PR DESCRIPTION
**Pull request by miron4dev**

### CODE REVIEW

File: core/src/main/java/com/contentful/vault/SyncConfig.java [hunk 1]
```diff
@@ -90,6 +92,17 @@ public Builder setSpaceId(String spaceId) {
       return this;
     }
 
+    public Builder setEnvironment(String environment) {
+      if (client != null) {
+        throw new IllegalStateException(
+            "Do not set an environment, when a client is already set. User either environment " +
```
Comment (JoseLion) [inline core/src/main/java/com/contentful/vault/SyncConfig.java]: @miron4dev I'm not sure I understand what this message means, is `User` the correct word at the start of the second sentence? Can you please explain what the message tries to say here?

File: core/src/main/java/com/contentful/vault/SyncConfig.java [hunk 2]
```diff
@@ -104,6 +117,13 @@ public Builder setClient(CDAClient client) {
         );
       }
 
+      if (environment != null) {
+        throw new IllegalStateException(
+            "Do not set a client, when an environment is already set. User either environment " +
```
Comment (JoseLion) [inline core/src/main/java/com/contentful/vault/SyncConfig.java]: @miron4dev same here. Also if we are throwing the same message I think we can extract the text to a constant, what do you think?

File: core/src/main/java/com/contentful/vault/SyncConfig.java (hunk 1 above)
Comment (miron4dev) [reply core/src/main/java/com/contentful/vault/SyncConfig.java]: You're right. It's a typo.

File: core/src/main/java/com/contentful/vault/SyncConfig.java (hunk 2 above)
Comment (miron4dev) [reply core/src/main/java/com/contentful/vault/SyncConfig.java]: Good idea. I'll do it tomorrow.

**Pull request merged**